### PR TITLE
Aman/unittests other changes

### PR DESCRIPTION
**Pull request by ataddesse**
image
image

This PR: 
- Adds test cases for leads.service
- Adds module files for each component to simplify testing and increase code coverage. (modules are not imported in app.module yet, only used for testing purpose now)

Include a list of changes made or features created. Include any relevant screenshots or links.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (change that introduces a bug or causes app to crash, but that we plan on fixing later)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Provide any test suites/filenames that you had to create/change.

- [ ] Test A
- [ ] Test B

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: frontend/src/app/components/crm/leads/leads.component.html [hunk 1]
```diff
@@ -119,11 +122,17 @@
...
+    <mat-card *ngIf="isLoading"
+              style="display: flex; justify-content: center; align-items: center">
+      <mat-progress-spinner
+        color="primary"
+        mode="indeterminate">
+      </mat-progress-spinner>
+    </mat-card>
     <!-- TODO: Switch this to use a class instead of setting styles -->
-    <mat-paginator [pageSizeOptions]="[5, 10, 20]"
+    <mat-paginator [pageSizeOptions]="[10, 5, 20, 30]"
```
Comment (FluffyPenguin) [inline frontend/src/app/components/crm/leads/leads.component.html:133]: Can we make this something like 25, 50, 100?

File: frontend/src/app/components/landing/landing.component.css [hunk 2]
```diff
@@ -12,3 +12,23 @@
...
+.login-display {
+    font-size: 40px;
+    width: 40px;
+  vertical-align: middle;
+  margin-bottom: 10px;
+}
+.icon-text {
+    display: flex;
+    align-items: center;
+}
```
Comment (FluffyPenguin) [inline frontend/src/app/components/landing/landing.component.css:34]: nit: fix tabbing

File: frontend/src/app/components/landing/landing.component.html [hunk 3]
```diff
@@ -6,11 +6,13 @@ <h6>miniCRM</h6>
...
             <span class="mat-display-4">
-            A simple, <br> 
+            A simple, <br>
             lightweight <br>
             CRM app. <br>
             </span>
-            <button (click)="login()" mat-raised-button color="primary">LOGIN</button>
+          <div class="icon-text">
+          <button (click)="login()" mat-raised-button class="material-cust-large-button" color="primary"> <span class="mat-display-1">Login </span><mat-icon class="login-display"> login</mat-icon></button>
+          </div>
```
Comment (FluffyPenguin) [inline frontend/src/app/components/landing/landing.component.html:15]: nit: tab the button line to the right
<div>
    <button>
</div>

File: frontend/src/app/components/landing/landing.component.css (hunk 2 above)
Comment (ataddesse) [reply frontend/src/app/components/landing/landing.component.css:34]: gotchu

File: frontend/src/app/components/landing/landing.component.html (hunk 3 above)
Comment (ataddesse) [reply frontend/src/app/components/landing/landing.component.html]: gotchu

File: frontend/src/app/components/crm/leads/leads.component.html (hunk 1 above)
Comment (ataddesse) [reply frontend/src/app/components/crm/leads/leads.component.html:133]: yep

**Pull request merged**